src/Serilog.Sinks.AzureDocumentDb/Sinks/AzureDocumentDb/AzureDocumentDbSink.cs

```diff
@@ -35,42 +36,37 @@ public AzureDocumentDBSink(Uri endpointUri, string authorizationKey, string data
             _formatProvider = formatProvider;
             _client = new DocumentClient(endpointUri, authorizationKey);
             _storeTimestampInUtc = storeTimestampInUtc;
-            Task.WaitAll(new []{CreateDatabaseIfNotExistsAsync(databaseName)});
-            Task.WaitAll(new []{CreateCollectionIfNotExistsAsync(collectionName)});
+            CreateDatabaseIfNotExistsAsync(databaseName).ConfigureAwait(false);
+            CreateCollectionIfNotExistsAsync(collectionName).ConfigureAwait(false);
         }
 
         public void Emit(LogEvent logEvent)
         {
-            Task.WaitAll(EmitAsync(logEvent));
+            EmitAsync(logEvent).ConfigureAwait(false);
```

Comment:
@nblumhardt appreciated for review. However I have a concern here, .Wait() is essentially killing purpose async as we'll be blocking calling thread until process completes. I believe this method should return as soon as possible and be available to next call.

This may not be an issue under light traffic but may be an issue when under stress. Let's say we are using Verbose as log level.

I'm suggesting to introduce a buffer here and some kind dedicated background task/thread takes responsibility of sending log to azure.